pipeline -> job symbol renames

smackesey opened the pull request:
Internal companion PR: https://github.com/dagster-io/internal/pull/5417

This is a very large PR that renames `pipeline` to `job` almost everywhere it is used except for module names (these are done in a separate upstack PR) and the GQL schema: in class names, function/method names, arguments, local variables, and docstrings. This is appropriate after folding `PipelineDefinition` into `JobDefinition` in #13327.

While the volume of changes is large, the engineering complexity is low. The changes were made by grepping for "pipeline" and using Pylance "Rename symbol" on any instance I came across. In some cases I also needed to manually follow up to convert untyped references and references in docstrings. In cases where I discovered an untyped reference, I sometimes added type annotations to the containing function as well. When renaming a serdes class or field, I added the appropriate `storage_name` or `storage_field_names` to`@whitelist_for_serdes` to keep the serialization format constant.

I have kept some structure in the commits so that this can be broken up if necessary, though I would like to avoid that due to this having an internal companion PR in which every split will need to be duplicated. Also, though there is structure to the commits (I tried to scope them to individual classes and functions-- see below), fixing errors on rebase has introduced some deviations to that structure, (e.g. if a rebase introduced a new callsite to some `*pipeline*` method, I needed to rename it in a different commit than the initial one commit that renamed all prior instances), so any split will require some additional work in interactive rebase slice-and-dice of commits where I try to corral all renames into the "structured" correct commit.

Aside from standard symbol renames, there was one "non-standard" change where `"pipeline_name"` was replaced with `"job_name"` in the logging tag keys.

Below is a list of all the renames of non-local symbols (the PR also included many local symbol renames). The list uses the following scheme:

- `Foo -> Bar`: a class or top-level function was renamed
- `Foo.*pipeline* -> *job*`: attributes, methods, and method arguments of Foo containing "pipeline" were renamed with "job" equivalent
- `foo(*pipeline*) -> foo(*job*)`: arguments of foo containing "pipeline" were renamed with "job" equivalent
- `Foo.*pipeline* -> Bar.*pipeline*`: class/top-level function was renamed AND attributes/methods/arguments renamed
- `locals`: assorted local variable and docstring renames.

Here is the list:

1. `PipelineRunStatsSnapshot -> DagsterRunStatsSnapshot`
1. `PipelinePythonOrigin -> JobPythonOrigin`
1. `DagsterRun.*pipeline* -> *job*`
1. `ExternalPipelineOrigin -> ExternalJobOrigin`
1. `PipelineSnapshot.*pipeline* -> JobSnapshot.*job*`
1. `PipelineSnapshotSerializer -> JobSnapshotSerializer`
1. `create_pipeline_snapshot_id -> create_job_snapshot_id`
1. `PipelineSnapshotLineage -> JobLineageSnapshot`
1. `ExecutionPlanSnapshotArgs.*pipeline* -> *job*`
1. `ExecuteRunArgs.*pipeline* -> *job*`
1. `ResumeRunArgs.*pipeline* -> *job*`
1. `ExecuteExternalPipelineArgs -> ExecuteExternalJobArgs`
1. `ExecuteExternalJobArgs.*pipeline* -> *job*`
1. `ExecuteStepArgs.*pipeline* -> *job*`
1. `PipelineIndex.*pipeline* -> JobIndex.*job*`
1. `RepresentedPipeline.*pipeline* -> RepresentedJob.*job*`
1. `HistoricalPipeline.*pipeline* -> HistoricalJob.*job*`
1. `PipelineFailureData -> JobFailureData`
1. `PipelineCanceledData -> JobCanceledData`
1. `IPipeline -> IJob`
1. `InMemoryPipeline.*pipeline* -> InMemoryJob.*job*`
1. `ExternalPipeline.*pipeline* -> ExternalJob.*job*`
1. `PipelineRunReaction -> DagsterRunReaction`
1. `ReconstructablePipeline.*pipeline* -> ReconstructableJob.*job*`
1. `get_ephemeral_repository_name(*pipeline*) -> *job*`
1. `ReconstructableRepository.*pipeline* -> *job*`
1. `bootstrap_standalone_recon_pipeline -> bootstrap_standalone_recon_job`
1. `ExternalRepositoryOrigin.*pipeline* -> *job*`
1. `PipelineSubsetSnapshotArgs -> JobSubsetSnapshotArgs`
1. `DagsterK8sPipelineStatusException -> DagsterK8sJobStatusException`
1. `ExternalPipelineSubsetResult -> ExternalJobSubsetResult`
1. `get_step_stats_by_key(*pipeline*) -> *job*`
1. `verify_step(*pipeline*) -> *job*`
1. `scaffold_pipeline_config -> scaffold_job_config`
1. `sync_get_external_pipeline_subset_grpc -> sync_get_external_job_subset_grpc`
1. `{RepoRelativeTarget,DirectTarget}.*pipeline* -> *job*`
1. `RepositoryDefinition.*pipeline* -> *job*`
1. `log_pipeline_event -> log_job_event`
1. `DagsterEvent.*pipeline* -> *job*`
1. `StepExecutionContext.*pipeline* -> *job*`
1. `IPlanContext.*pipeline* -> *job*`
1. `PlanData.*pipeline* -> *job*`
1. `InitExecutorContext.*pipeline* -> *job*`
1. `ContextCreationData.*pipeline* -> *job*`
1. `RepositoryPythonOrigin.*pipeline* -> *job*`
1. `ExternalSchedule.*pipeline* -> *job*`
1. `ExternalSensor.*pipeline* -> *job*`
1. `ExternalPartitionSet.*pipeline* -> *job*`
1. `ExternalSensorData.*pipeline* -> *job*`
1. `RunStatusSensorDefinition.*pipeline* -> *job*`
1. `pipeline_def_from_pointer -> job_def_from_pointer`
1. `JobDefinition.*pipeline* -> *job*`
1. `ExternalRepositoryData.*pipeline* -> *job*`
1. `external_job_data_from_def(*pipeline*) -> *job*`
1. `CodeLocation.*pipeline* -> *job*`
1. `build_node_defs_snapshot(*pipeline*) -> *job*`
1. `build_config_schema_snapshot(*pipeline*) -> *job*`
1. `StepInputSource.*pipeline* -> *job*`
1. `FromConfig.*pipeline* -> *job*`
1. `HookContext.*pipeline* -> *job*`
1. `InputContext.*pipeline* -> *job*`
1. `create_context_creation_data(*pipeline*) -> *job*`
1. `UnboundOpExecutionContext.*pipeline* -> *job*`
1. `InitLoggerContext.*pipeline* -> *job*`
1. `OutputContext.*pipeline* -> *job*`
1. `DagsterInstance.*pipeline* -> *job*`
1. `pipeline_execution_iterator -> job_execution_iterator`
1. `generate_dep_graph(*pipeline*) -> *job*`
1. `{LaunchRunContext,ResumeRunContext}.*pipeline* -> *job*`
1. `ComputeLogManager.*pipeline* -> *job*`
1. `RunStorage.*pipeline* -> *job*`
1. `LocalComputeLogManager.*pipeline* -> *job*`
1. `recon_pipeline_from_origin -> recon_job_from_origin`
1. `external_pipeline_from_recon_pipeline -> external_job_from_recon_job`
1. `check_cli_execute_file_pipeline -> check_cli_execute_file_job`
1. `build_init_logger_context(*pipeline*) -> *job*`
1. `{print_{ops,op,job}(*pipeline*) -> *job*`
1. `DELETE test_logger_{pipeline,both}_def`
1. `sync_get_external_execution_plan_grpc(*pipeline*) -> *job*`
1. `sync_get_external_job_subsest_grpc(*pipeline*) -> *job*`
1. `RunConfigSchemaCreationData.*pipeline* -> *job*`
1. `create_run_config_schema(*pipeline*) -> *job*`
1. `PipelineSelector.*pipeline* -> JobSubsetSelector.*job*`
1. `PipelineSelector -> JobSubsetSelector`
1. `external_asset_graph_from_defs(*pipeline*) -> *job*`
1. `execute_step_out_of_process(*pipeline*) -> *job*`
1. `create_run_for_job(*pipeline*) -> *job*`
1. `ExecutionPlanSnapshot.*pipeline* -> *job*`
1. `ExecutionStep.*pipeline* -> *job*`
1. `UnresolvedMappedExecutionStep.*pipeline* -> *job*`
1. `UnresolvedCollectExecutionStep.*pipeline* -> *job*`
1. `ExecutionPlan.*pipeline* -> *job*`
1. `get_output_context(*pipeline*) -> *job*`
1. `step_output_version(*pipeline*) -> *job*`
1. `execution_context_event_generator(*)`
1. `PlanOrchestrationContext.*pipeline* -> *job*`
1. `orchestration_context_event_generator(*pipeline*) -> *job*`
1. `PlanExecutionContextManager.*pipeline* -> *job*`
1. `scoped_pipeline_context(*pipeline*) -> *job*`
1. `DebugRunPayload.*pipeline* -> *job*`
1. `RunsFilter.*pipeline* -> *job*`
1. `DagsterLoggingMetadata.*pipeline* -> *job*`
1. `log_repo_stats(*pipeline*) -> *job*`
1. `ResolvedRunConfig.build(*pipeline*) -> *job*`
1. `config_map_loggers(*pipeline*) -> *job*`
1. `create_run_for_test(*pipeline*) -> *job*`
1. `SubmitRunContext.*pipeline* -> *job*`
1. `parse_solid_selection(*pipeline*) -> *job*`
1. `core_execute_run(*pipeline*) -> *job*`
1. `CodeLocation.get_external_execution_plan(*pipeline*) -> *job*`
1. `external_job_from_location(*pipeline*) -> *job*`
1. `_grpc.impl.core_execute_run(*pipeline*) -> *job*`
1. `core_execute_in_process(*pipeline*) -> *job*`
1. `execute_host_run_mode(*pipeline*) -> *job*`
1. `execute_run_iterator(*pipeline*) -> *job*`
1. `execute_plan(*pipeline*) -> (*job*)`
1. `create_execution_plan(*pipeline*) -> *job*`
1. `context.get_external_execution_plan`
1. `ExternalJobRef.*pipeline* -> *job*`
1. `StepHandlerContext.*pipeline* -> *job*`
1. `StepRunRef.*pipeline* -> *job*`
1. `execute_plan_iterator(*pipeline*) -> *job*`
1. `execute_plan(*pipeline*) -> *job*`
1. `logging_tags["pipeline"] -> "job"`
1. `EventLogEntry.*pipeline* -> *job*`
1. `JobInfo.*pipeline* -> *job*`
1. `DagstermillExecutionContext.*pipeline* -> *job*`
1. `SensorExecutionData.*pipeline* -> *job*`
1. `ExecuteRunWithPlanIterable.*pipeline* -> *job*`
1. `GraphExecutionResult.*pipeline* -> *job*`
1. `MultiprocessExecutorChildProcessCommand.*pipeline* -> *job*`
1. `PipelineExecutionResult.*pipeline* -> *job*`
1. `CompositeSolidExecutionResult.*pipeline* -> *job*`
1. `OpExecutionResult.*pipeline* -> *job*`
1. `RunFailureSensorContext.*pipeline* -> *job*`
1. `AbstractComputeExecutionContext.*pipeline* -> *job*`
1. `ExternalTargetData.*pipeline* -> *job*`
1. `RunStorage.*pipeline* -> *job*`
1. `dagstermill.Manager.*pipeline* -> *job*`
1. `inner_plan_execution_iterator(*pipeline*) -> *job*`
1. `log_dagster_event(*pipeline*) -> *job*`
1. `scoped_pipeline_context -> scoped_job_context`
1. `log_job_event(*pipeline*) -> *job*`
1. `ActiveExecution.*pipeline* -> *job*`
1. `get_pipeline_subset_or_error_from_pipeline_selector(*pipeline*) -> get_job_subset_or_error_from_job_selector(*job*)`
1. `get_job_snapshot_or_error_from_snapshot_id -> get_job_snapshot_or_error_from_snapshot_id`
1. `get_pipeline_reference_or_raise(*pipeline*) -> get_job_reference_or_raise(*job*)`
1. `get_pipeline_from_selector -> get_job_from_selector`
1. `EcsContainerContext.*pipeline* -> *job*`
1. `get_pipeline_or_error -> get_job_or_error`
1. `get_full_external_pipeline_or_raise -> get_full_external_job_or_raise`
1. `get_external_pipeline_or_raise -> get_external_job_or_raise`
1. `ensure_valid_config(*pipeline*) -> *job*`
1. `RunStatusSensorContext.*pipeline* -> *job*`
1. `create_run_for_test`

smackesey commented:
Current dependencies on/for this PR:
* master
  * **PR #13145** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13145" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
    * **PR #13108** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13108" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
      * **PR #13168** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13168" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
        * **PR #13169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13169" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
          * **PR #13081** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13081" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
            * **PR #13083** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13083" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
            * **PR #12770** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12770" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
              * **PR #12769** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12769" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
              * **PR #13170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13170" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                * **PR #13171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13171" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                  * **PR #13188** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13188" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                    * **PR #13172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13172" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                      * **PR #13192** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13192" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                        * **PR #13301** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13301" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                        * **PR #13295** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13295" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                          * **PR #13296** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13296" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                            * **PR #13294** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13294" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                              * **PR #13297** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13297" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                * **PR #13303** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13303" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                  * **PR #13299** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13299" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                    * **PR #13298** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13298" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                      * **PR #13302** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13302" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                        * **PR #13311** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13311" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                          * **PR #13312** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13312" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                            * **PR #13316** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13316" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                              * **PR #13317** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13317" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                                * **PR #13318** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13318" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                                  * **PR #13321** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13321" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                                    * **PR #13320** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13320" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                                      * **PR #13327** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13327" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                                                        * **PR #13497** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13497" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a>  👈
                                                        * **PR #13496** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13496" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 
                      * **PR #13173** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13173" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13497?utm_source=stack-comment).

dpeng817 commented:
Nice. I kind of wish this was done in a more class-by-class basis, but I understand how nightmarish a task that would actually be. 
I think the important things to look out for here are:
- No accidental renames to serdes storage names
- No accidental removal of APIs that haven't had a replacement until very recently
- Nonsensical renames as a result of commit thrash / rebasing.
@alangenfeld anything else you can think of there? My plan here is to comb through every file over the course of the next few days essentially.

dpeng817 commented on integration_tests/python_modules/dagster-k8s-test-infra/dagster_k8s_test_infra/integration_utils.py:
```diff
@@ -140,22 +141,22 @@ def _execute_query_over_graphql(dagit_url, query, variables):
                     "repositoryLocationName": code_location_name,
                     "repositoryName": repository_name,
-                    "pipelineName": pipeline_name,
+                    "pipelineName": job_name,
```
does pipelineName need to get kept around here for backcompat purposes?

smackesey replied:
I didn't touch the GQL schema in general, since that's a whole separate project requiring updates to the queries used in dagit

smackesey commented on python_modules/dagster-graphql/dagster_graphql/schema/partition_sets.py:
```diff
@@ -165,7 +165,12 @@ class GraphenePartition(graphene.ObjectType):
         name = "Partition"
 
-    def __init__(self, external_repository_handle, external_partition_set, partition_name):
+    def __init__(
+        self,
+        external_repository_handle: RepositoryHandle,
+        external_partition_set: ExternalPartitionSet,
+        partition_name: str,
+    ):
```
Not much, it's basically in the same situation as `mode` in the GQL schema. But this stuff is still referenced in dagit so we can't delete it yet. There will be a follow-up removal of all of this stuff from GQL/dagit.

dpeng817 reviewed (commented):
Some process oriented feedback:
I'm finding this PR a challenge to review because of its multifaceted nature. This PR includes renames, type annotations, and serdes changes - I think with renames and type annotations we can generally trust our CICD to catch most things, and so I think review on those individual pieces would be quick. Serdes changes however, even just a simple rename, I generally just don't feel super confident in our tests catching everything. I'm not convinced that all of the serdes objects are under test well enough to ensure that backcompat is always preserved. I think the amalgam of these three things leaves me feeling unconfident in relying on our CICD to catch any errors that might exist - not to mention typos, rebase artifacts etc. All in all, I do actually feel the need to comb every file of this PR, which is taking a while.

I think I understand why this PR is as big as it is, this is certainly a large undertaking. However for future rename suites (I assume a solid one is coming down the pipe) I'd have the following suggestions so that we can have reviews happen quicker:
- Separate out any serdes renames into their own PR, and maybe mark in github via comments where those renames are occurring so that I can easily find them.
- Don't include type annotations - I understand that it's net good for the codebase, but I just find it to muddle things when scanning through

In any case, great job on getting this done - just some suggestions for streamlining review of stuff like this in the future

dpeng817 commented on python_modules/dagster-graphql/dagster_graphql_tests/graphql/snapshots/snap_test_graph.py:
```diff
@@ -7,6 +7,111 @@
 
 snapshots = Snapshot()
 
+snapshots['TestGraphs.test_basic_graphs[non_launchable_postgres_instance_lazy_repository] 1'] = {
```
I guess these snapshot changes are ok since they are the result of a rename + are purely additive

dpeng817 commented on python_modules/dagster/dagster/_core/definitions/reconstruct.py:
```diff
@@ -163,67 +163,72 @@ def get_python_origin_id(self) -> str:
 
-@whitelist_for_serdes
-class ReconstructablePipeline(
+@whitelist_for_serdes(
+    storage_name="ReconstructableJob",
+    storage_field_names={
+        "job_name": "pipeline_name",
+    },
+)
+class ReconstructableJob(
```
shouldn't the storage name here be "ReconstructablePipeline"? I guess maybe it's fine that it's reconstructable job, as it should only ever be used within user code, which should all be running the same version of dagster? @alangenfeld thoughts?

smackesey replied:
You're right, that's a legit mistake

dpeng817 commented on python_modules/dagster/dagster/_core/definitions/reconstruct.py:
```diff
@@ -335,60 +340,55 @@ def subset_for_execution_from_existing_pipeline(
+    function that returns a :py:class:`~dagster.JobDefinition`/:py:class:`~dagster.JobDefinition`,
     or a function decorated with :py:func:`@job <dagster.job>`.
 
-    When your pipeline/job must cross process boundaries, e.g., for execution on multiple nodes or
-    in different systems (like ``dagstermill``), Dagster must know how to reconstruct the pipeline/job
```
so satisfying to see these  / names go away

dpeng817 commented on python_modules/dagster/dagster/_core/events/__init__.py:
```diff
@@ -352,14 +352,18 @@ def handle_unpack_error(
 
 @whitelist_for_serdes(
-    serializer=DagsterEventSerializer, storage_field_names={"node_handle": "solid_handle"}
+    serializer=DagsterEventSerializer,
+    storage_field_names={
+        "node_handle": "solid_handle",
+        "job_name": "pipeline_name",
+    },
```
this looks fine

dpeng817 commented on python_modules/dagster/dagster/_core/events/__init__.py:
```diff
@@ -1574,32 +1568,32 @@ def engine_error(error: SerializableErrorInfo) -> "EngineEventData":
 
-@whitelist_for_serdes
-class PipelineFailureData(
+@whitelist_for_serdes(storage_name="PipelineFailureData")
+class JobFailureData(
```
[serdes rename] looks right

dpeng817 commented on python_modules/dagster/dagster/_core/events/__init__.py:
```diff
@@ -1574,32 +1568,32 @@ def engine_error(error: SerializableErrorInfo) -> "EngineEventData":
 
-@whitelist_for_serdes
-class PipelineCanceledData(
+@whitelist_for_serdes(storage_name="PipelineCanceledData")
+class JobCanceledData(
```
[serdes rename] looks right

dpeng817 commented on python_modules/dagster/dagster/_core/events/log.py:
```diff
@@ -27,6 +26,7 @@
     # machinery from events that are yielded by user code
     old_storage_names={"DagsterEventRecord", "LogMessageRecord", "EventRecord"},
     old_fields={"message": ""},
+    storage_field_names={"job_name": "pipeline_name"},
```
[serdes rename] looks right

dpeng817 commented on python_modules/dagster/dagster/_core/execution/context/invocation.py:
just wanted to ensure that these renames only replace things that didn't already exist / no deletion is required here

smackesey replied:
confirmed

dpeng817 commented on python_modules/dagster/dagster/_core/host_representation/external_data.py:
```diff
@@ -224,60 +224,68 @@ def get_external_sensor_data(self, name):
 
-@whitelist_for_serdes
-class ExternalPipelineSubsetResult(
+@whitelist_for_serdes(
+    storage_name="ExternalPipelineSubsetResult",
+    storage_field_names={"external_job_data": "external_pipeline_data"},
+)
+class ExternalJobSubsetResult(
```
[serdes rename] seems reasonable

dpeng817 commented on python_modules/dagster/dagster/_core/host_representation/external_data.py:
```diff
@@ -224,60 +224,68 @@ def get_external_sensor_data(self, name):
 
-@whitelist_for_serdes(old_fields={"is_job": True})
-class ExternalPipelineData(
+@whitelist_for_serdes(
+    storage_name="ExternalPipelineData",
+    storage_field_names={
+        "job_snapshot": "pipeline_snapshot",
+        "parent_job_snapshot": "parent_pipeline_snapshot",
+    },
+    old_fields={"is_job": True},
+)
```
[serdes rename] seems reasonable

dpeng817 replied:
maybe add a comment here for historical context - explaining why the `is_job` stuff is necessary

dpeng817 commented on python_modules/dagster/dagster/_core/host_representation/external_data.py:
```diff
@@ -371,14 +379,16 @@ def __new__(
 
 
-@whitelist_for_serdes(skip_when_empty_fields={"default_status"})
+@whitelist_for_serdes(
+    storage_field_names={"job_name": "pipeline_name"}, skip_when_empty_fields={"default_status"}
+)
```
[serdes rename] seems reasonable

dpeng817 commented on python_modules/dagster/dagster/_core/host_representation/external_data.py:
```diff
@@ -435,17 +445,17 @@ def __new__(cls, error: Optional[SerializableErrorInfo]):
 
 
-@whitelist_for_serdes
+@whitelist_for_serdes(storage_field_names={"job_name": "pipeline_name"})
```
[serdes rename] seems reasonable

dpeng817 commented on python_modules/dagster/dagster/_core/host_representation/external_data.py:
```diff
@@ -468,13 +478,16 @@ def __new__(cls, asset_keys: Optional[Sequence[AssetKey]] = None):
 
 
-@whitelist_for_serdes(skip_when_empty_fields={"default_status", "sensor_type"})
+@whitelist_for_serdes(
+    storage_field_names={"job_name": "pipeline_name"},
+    skip_when_empty_fields={"default_status", "sensor_type"},
+)
```
[serdes rename] seems reasonable

dpeng817 commented on python_modules/dagster/dagster/_core/origin.py:
```diff
@@ -67,25 +67,28 @@ def __new__(
 
-@whitelist_for_serdes
-class PipelinePythonOrigin(
+@whitelist_for_serdes(
+    storage_name="PipelinePythonOrigin",
+    storage_field_names={"job_name": "pipeline_name"},
+)
+class JobPythonOrigin(
```
[serdes rename] reasonable

dpeng817 commented on python_modules/dagster/dagster/_core/snap/execution_plan_snapshot.py:
```diff
@@ -33,14 +33,17 @@ def create_execution_plan_snapshot_id(execution_plan_snapshot: "ExecutionPlanSna
 
 
-@whitelist_for_serdes(skip_when_empty_fields={"repository_load_data"})
+@whitelist_for_serdes(
+    storage_field_names={"job_snapshot_id": "pipeline_snapshot_id"},
+    skip_when_empty_fields={"repository_load_data"},
+)
```
[serdes rename] seems fine

dpeng817 commented on python_modules/dagster/dagster/_core/storage/pipeline_run.py:
```diff
@@ -96,10 +96,10 @@ class DagsterRunStatus(Enum):
 
-@whitelist_for_serdes
-class PipelineRunStatsSnapshot(
+@whitelist_for_serdes(storage_name="PipelineRunStatsSnapshot")
+class DagsterRunStatsSnapshot(
```
[serdes rename] seems reasonable

dpeng817 commented on python_modules/dagster/dagster/_grpc/types.py:
```diff
@@ -10,25 +10,30 @@
 
 
-@whitelist_for_serdes
+@whitelist_for_serdes(
+    storage_field_names={
+        "job_origin": "pipeline_origin",
+        "job_snapshot_id": "pipeline_snapshot_id",
+    }
+)
```
[serdes rename] looks reasonable

dpeng817 commented on python_modules/dagster/dagster/_grpc/types.py:
```diff
@@ -70,42 +73,47 @@ def __new__(
 
 
-@whitelist_for_serdes
+@whitelist_for_serdes(
+    storage_field_names={
+        "job_origin": "pipeline_origin",
+        "run_id": "pipeline_run_id",
+    }
+)
```
[serdes rename] looks reasonable

dpeng817 commented on python_modules/dagster/dagster/_grpc/types.py:
```diff
@@ -116,41 +124,46 @@ def __new__(
 
 
-@whitelist_for_serdes
+@whitelist_for_serdes(
+    storage_field_names={
+        "job_origin": "pipeline_origin",
+        "run_id": "pipeline_run_id",
+    }
+)
```
[serdes rename] looks reasonable

dpeng817 commented on python_modules/dagster/dagster/_grpc/types.py:
```diff
@@ -161,50 +174,61 @@ def __new__(
 
-@whitelist_for_serdes
-class ExecuteExternalPipelineArgs(
+@whitelist_for_serdes(
+    storage_name="ExecuteExternalPipelineArgs",
+    storage_field_names={
+        "job_origin": "pipeline_origin",
+        "run_id": "pipeline_run_id",
+    },
+)
```
[serdes rename] looks reasonable

dpeng817 commented on python_modules/dagster/dagster/_grpc/types.py:
```diff
@@ -161,50 +174,61 @@ def __new__(
 
 
-@whitelist_for_serdes
+@whitelist_for_serdes(
```
[serdes rename] looks reasonable

dpeng817 commented on python_modules/dagster/dagster/_grpc/types.py:
```diff
@@ -442,28 +464,31 @@ def __new__(
 
-@whitelist_for_serdes
-class PipelineSubsetSnapshotArgs(
+@whitelist_for_serdes(
+    storage_name="PipelineSubsetSnapshotArgs",
+    storage_field_names={
+        "job_origin": "pipeline_origin",
+    },
+)
+class JobSubsetSnapshotArgs(
```
[serdes rename] looks reasonable

dpeng817 reviewed (commented):
I've gone through everything - just a few things left to clear up, which I've added comments for, and I think this is good to go.

smackesey commented:
Alright, I believe I've addressed all comments. From the latest round:

- `RepresentedJob.storage_name` should be `RepresentedPipeline`
- Ensure renames in `context/invocation` weren't clobbering existing methods
- Add comment explaining `is_job` on `ExternalPipelineData`

Thanks for reviewing-- I know this was a bear.

To address some of your general feedback above, I agree I shouldn't have included the type annotations-- I ended up adding them when I was grepping for untyped references, since the lack of them was why a one-shot rename didn't work, but yeah probably best not to mix them in in the future.

More generally, the main reason I wanted to keep this one giant PR is because of the arduous process for handling parallel OSS/internal PRs-- every split containing a rename of a symbol appearing in both OSS and internal requires a corresponding split in internal, and maintaining parallel stacks across repos is very tedious with rebase errors and our relatively high rate of CI flakes-- every new rebase tends to require editing a few branches in the middle of a stack. If there had been a separate PR per serdes-rename here, that alone would have ballooned this to ~30 additional PRs, assuming ~20 serdes renames with 50% also appearing in internal.

So thanks for soldiering through!

dpeng817 commented:
@smackesey ya to be clear I wouldn't expect every serdes rename to be its own PR - but just to isolate the serdes renames to a PR specifically - I guess what could also work is organizing commits in that way - IE only serdes in one commit

dpeng817 reviewed (approved):
🙏
